### PR TITLE
Add NULL dereference checks

### PR DESCRIPTION
**Pull request by ttins**

### CODE REVIEW

File: iothub_client/src/message_queue.c [hunk 1]
```diff
@@ -54,7 +54,15 @@ static bool find_item_by_message_ptr(LIST_ITEM_HANDLE list_item, const void* mat
     MESSAGE_QUEUE_ITEM* current_item = (MESSAGE_QUEUE_ITEM*)singlylinkedlist_item_get_value(list_item);
     MQ_MESSAGE_HANDLE* target_item = (MQ_MESSAGE_HANDLE*)match_context;
 
-    return (current_item->message == target_item);
+    if (current_item == NULL)
```
Comment (danewalton) [inline iothub_client/src/message_queue.c]: the style in this doc (and I think most of this SDK) is to have one return point. So please update to have something like 
```c
bool result;

if()
{
    result = false;
}
else
{
//...
}

return result;
```

File: serializer/src/codefirst.c [hunk 2]
```diff
@@ -560,6 +560,12 @@ EXECUTE_COMMAND_RESULT CodeFirst_InvokeAction(DEVICE_HANDLE deviceHandle, void*
         result = EXECUTE_COMMAND_ERROR;
         LogError("parameterValues error %s ", MU_ENUM_TO_STRING(EXECUTE_COMMAND_RESULT, result));
     }
+    /*Codes_SRS_CODEFIRST_99_200:[ If devicHeader (callbackUserContext) is NULL then EXECUTE_COMMAND_ERROR shall be returned.]*/
```
Comment (danewalton) [inline serializer/src/codefirst.c]: typo: deviceHeader

Comment (ttins) [reply serializer/src/codefirst.c]: whoops good catch

File: iothub_client/src/message_queue.c (hunk 1 above)
Comment (ttins) [reply iothub_client/src/message_queue.c]: else:
result = (current_item->message == target_item);

**Pull request merged**